python/ray/serve/examples/tutorials/sklearn+composition.rst

```diff
@@ -0,0 +1,344 @@
+with Ray Serve - how to compose multiple models together using
+RayServeHandle
+
+`scikit-learn <https://scikit-learn.org/>`__ is a popular library with
+easy to use classification and regression models. In this tutorial, we
+will train several models to tackle a simple classification task with
+the iris dataset and compose them together.
+
+.. code:: ipython3
+
+    # Install dependencies
+    !pip install scikit-learn numpy ray
+
+Part 1: Training Classifers
+---------------------------
+
+We will be using the classical ``iris`` dataset, consisting of 150
+samples and 3 categories.
+
+Let’s train a models and validate its accuracy. We will use gradient
+boosted trees to start.
+
+.. code:: ipython3
+
+    from sklearn.datasets import load_iris
+    from sklearn.ensemble import GradientBoostingClassifier
+    from sklearn.metrics import mean_squared_error
+    
+    import numpy as np
+
+.. code:: ipython3
+
+    # Load data
+    data, target, target_names, description, feature_names, _ = load_iris().values()
+    
+    # Instantiate model
+    model = GradientBoostingClassifier()
+    
+    # Training and validation split
+    np.random.shuffle(data), np.random.shuffle(target)
+    train_x, train_y = data[:100], target[:100]
+    val_x, val_y = data[100:], target[100:]
+    
+    # Train and evaluate models
+    model.fit(train_x, train_y)
+    print("MSE:", mean_squared_error(model.predict(val_x), val_y))
+
+
+.. parsed-literal::
+
+    MSE: 1.38
+
+
+Part 2: Save the model and deploy to RayServe
+---------------------------------------------
+
+In this part, we will deploy the model to RayServe and issue queries.
+
+.. code:: ipython3
+
+    from ray import serve
+    
+    serve.init()
+
+
+.. parsed-literal::
+
+    2020-04-08 18:59:16,795	INFO resource_spec.py:212 -- Starting Ray with 33.06 GiB memory available for workers and up to 0.09 GiB for objects. You can adjust these settings with ray.init(memory=<bytes>, object_store_memory=<bytes>).
+    2020-04-08 18:59:17,201	INFO services.py:1148 -- View the Ray dashboard at [1m[32mlocalhost:8265[39m[22m
+
+
+.. parsed-literal::
+
+    [2m[36m(pid=37037)[0m INFO:     Started server process [37037]
+    [2m[36m(pid=37037)[0m INFO:     Waiting for application startup.
+    [2m[36m(pid=37037)[0m INFO:     Application startup complete.
+
+
+.. code:: ipython3
+
+    # Save the model and label to file
+    import pickle
+    import json
+    
+    with open("/tmp/iris_model_logistic_regression.pkl", "wb") as f:
+        pickle.dump(model, f)
+    with open("/tmp/iris_labels.json", "w") as f:
+        json.dump(target_names.tolist(), f)
+
+Our models are now available as files in temp folder.
+
+.. code:: ipython3
+
+    !ls /tmp/iris_model_logistic_regression.pkl
+
+
+.. parsed-literal::
+
+    /tmp/iris_model_logistic_regression.pkl
+
+
+Let’s first deploy the logistic regression model
+
+.. code:: ipython3
+
+    @serve.route("/logistic_regressor")
```

Comment:
Is this eagerly creating endpoints and connecting to backend? I cannot find create_backend & create endpoint & link when I read code. 

Also, I think it will be easier for users to understand if we focus on just one way to deploy services (either just decorator based routing or the traditional way.)